### PR TITLE
[HIPIFY] Added directory and file exclusion, updated whitelist

### PR DESCRIPTION
**Pull request by bjoo**
`-exclude-dirs=dirlist` will set up a list of directories to be excluded.
`-exclude-files=filelist` will set up a list of files to be excluded.

List separator is the `,` (comma) like for whitespaces.

Populate `whitelist` with a few more APIs.

### CODE REVIEW

**Review by emankov — CHANGES_REQUESTED** [review]
LGTM, otherwise formatting in the generated perl code.

File: src/CUDA2HIP_Perl.cpp [hunk 1]
```diff
@@ -411,94 +454,103 @@ namespace perl {
...
-    *streamPtr.get() << tab_2 << my << "$infile;" << endl;
-    *streamPtr.get() << tab_2 << my << "$outfile;" << endl;
-    *streamPtr.get() << tab_2 << "if (-e $file_prehip) {" << endl;
-    *streamPtr.get() << tab_3 << "$infile  = $file_prehip;" << endl;
-    *streamPtr.get() << tab_3 << "$outfile = $fileName;" << endl;
+    *streamPtr.get() << tab << "$fileDir=dirname( $fileName );" << endl;
+    *streamPtr.get() << tab << "if ( $exclude_dirhash{ $fileDir } ) { " << endl;
+    *streamPtr.get() << tab_2 <<  print << "\" Skipping file: $fileName in excluded directory $fileDir \\n\";" << endl_tab << "}";
+    *streamPtr.get() << tab << "if ( $exclude_filehash{ $fileName } ) { " << endl;
+    *streamPtr.get() << tab_2 <<  print << "\" Skipping  excluded file: $fileName \\n\";" << endl_tab << "}";
```
[the remark this answers is not in the document]
Comment (bjoo) [reply src/CUDA2HIP_Perl.cpp]: Added

**Comment by bjoo** [comment]
Requests completed.

**Review by emankov — APPROVED** [review]
Thank you!

**Comment by bjoo** [comment]
Hi @emankov . Since I have changed employment from ORNL to industry, I probably am not able to contribute to ROCm anymore.  Please reassign this.

**Comment by emankov** [comment]
> Hi @emankov . Since I have changed employment from ORNL to industry, I probably am not able to contribute to ROCm anymore. Please reassign this.

No problem, my friend. Thank you for all your previous contributions!

**Pull request merged**